exist-core/src/test/java/org/exist/xquery/update/UpdateInsertTriggersDefrag.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * eXist-db Open Source Native XML Database
+ * Copyright (C) 2001 The eXist-db Authors
+ *
+ * info@exist-db.org
+ * http://www.exist-db.org
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+package org.exist.xquery.update;
+
+import org.exist.storage.DBBroker;
+import org.exist.test.ExistXmldbEmbeddedServer;
+import org.exist.test.TestConstants;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.xmldb.api.base.Collection;
+import org.xmldb.api.base.ResourceSet;
+import org.xmldb.api.base.XMLDBException;
+import org.xmldb.api.modules.CollectionManagementService;
+import org.xmldb.api.modules.XMLResource;
+import org.xmldb.api.modules.XPathQueryService;
+import org.xmldb.api.modules.XQueryService;
+
+import static org.exist.util.PropertiesBuilder.propertiesBuilder;
+import static org.junit.Assert.assertEquals;
+
+public class UpdateInsertTriggersDefrag {
+    @ClassRule
+    public static final ExistXmldbEmbeddedServer exist = new ExistXmldbEmbeddedServer(false, true, true, propertiesBuilder().put(DBBroker.PROPERTY_XUPDATE_FRAGMENTATION_FACTOR, -1).build());
+    final String path = TestConstants.TEST_COLLECTION_URI + "/" + TestConstants.TEST_XML_URI.toString();
+    final String xml = "<list><item>initial</item></list>";
+    Collection testCollection;
+    XQueryService queryService;
+    CollectionManagementService collectionService;
+
+    /**
+     * stores XML String and get Query Service
+     *
+     * @param documentName to be stored in the DB
+     * @param content      to be stored in the DB
+     * @throws XMLDBException if an error occurs storing the document
+     */
+    private void storeXML(final String documentName, final String content) throws XMLDBException {
+        final XMLResource doc = testCollection.createResource(documentName, XMLResource.class);
```

Comment:
1. The XML Resource has not been closed after it is finished with, so this test may leak resources.